### PR TITLE
DL3-3 Added a React UI for Deep Linking

### PR DESCRIPTION
**Pull request by mpellicer**
## Description
This PR includes all the work described in https://lumenlearning.atlassian.net/browse/DL3-3.

Incorporates an additional set of UI templates written in React for deep linking requests, this does not affect other type of requests like resources.

The UI displays the new waymaker course picker with some static data until the endpoints are ready to be integrated.

### Motivation and Context
When the user wants to link a Waymaker course through the LMS, via Deep Linking, the LMS sends an LTI 1.3 Deep Linking request, it redirects to the React UI, which contains the new designs of the Waymaker deep linking UI.

### Fixes
[Jira ticket](https://lumenlearning.atlassian.net/browse/DL3-3)

## How Has This Been Tested?
This has been tested in the Unicon's D2L instance using a local development environment, redirecting all the endpoints to the Middleware using Ngrok. 

A proper test plan could be (Not including the LTI deployment stuff):

- As an instructor or admin, go to a course.
- Go to the content tool.
- Add an Existing Activity.
- Select the tool link of the LTI tool.
- Ensure the Waymaker course picker is displayed in the modal.

## Screenshots
image

image
main
course-info

## Database changes
No DB changes.

## ⚠️ Deployment instructions ⚠️
No additional instructions.

## New ENV variables
No new ENV variables needed.
The deep linking option now is enabled by default.

```
lti13.enableDeepLinking=true
```

---

## Checklist

- [X] I have reviewed the AC for this feature and my changes meet all requirements
- [X] (For UI changes) I have considered the accessibility of this feature (see https://www.a11yproject.com/checklist/ for a high-level checklist)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have requested a review from at least one other dev

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -37,7 +37,7 @@ lti13.demoMode=false
 lti13.enableAGS=false
 lti13.enableMembership=false
 lti13.enableDynamicRegistration=true
-lti13.enableDeepLinking=false
+lti13.enableDeepLinking=true
```
Comment (mgwozdz-unicon) [inline src/main/resources/application.properties:40]: Could you please make this change in application-aws.properties as well?

Comment (mpellicer) [reply src/main/resources/application.properties:40]: Of course, done.

**Review by LeftShotDev — APPROVED** [review]
The React/ Redux implementation here looks great, easy to follow what is happening on my first time looking at this code base.

I do not have this repo setup to run locally, so I did not test it on my local machine.

File: src/main/frontend/src/app/staticData.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const releaseDate = '06/03/2022';
+
+
+const sectionArray = [
+  {name: 'Course Contents', items: ['About This Course.', 'Course Contents at a Glance.', 'Course Learning Outcomes.']},
+  {name: 'Course Contents', items: ['About This Course.', 'Course Contents at a Glance.', 'Course Learning Outcomes.']},
+  {name: 'Course Contents', items: ['About This Course.', 'Course Contents at a Glance.', 'Course Learning Outcomes.']},
+];
+
+export const COURSE_ARRAY = [
```
Comment (smith750) [inline src/main/frontend/src/app/staticData.js:12]: I'll make a note about this ini the channel.

**Review by smith750 — APPROVED** [review]
This looks good to me but I'm making a comment in the Slack channel about one concern I have.

File: src/main/frontend/src/app/staticData.js (hunk 2 above)
Comment (mpellicer) [reply src/main/frontend/src/app/staticData.js:12]: @smith750 This is just a mock object, I'll replace the object by the schema when the endpoint is available, thanks!

**Comment by mpellicer** [comment]
Shall I care about the Travis CI build? It is failing but I suspect is not related to the changes of this PR. 

If everybody is fine could you please merge it? I don't have permissions to do it.

**Comment by smith750** [comment]
I merged the PR.  We'll look into the build failure.

**Pull request merged**